### ISSUE TITLE
LKML hot patches

### ISSUE THREAD

**Issue by xuliker** [xuliker/LKML-email-pure-viewer#38]
* kunit: introduce KUnit, the Linux kernel unit testing framework
  * https://lkml.org/lkml/2018/10/24/185
* 86/mm/tlb: make lazy TLB mode even lazier
  * https://lkml.org/lkml/2018/9/24/1573
* x86: dynamic indirect call promotion
  * https://lkml.org/lkml/2018/10/18/175
* Intel Processor Trace virtualization enabling
  * https://lkml.org/lkml/2017/10/16/1111
* improve vmalloc allocation
  * https://lkml.org/lkml/2018/10/19/786
* kernel/signal: Signal-based pre-coredump notification
  * https://lkml.org/lkml/2018/10/13/47
* Allow persistent memory to be used like normal RAM
  * https://lkml.org/lkml/2018/10/23/9
* [ANNOUNCE] iproute2 4.19
  - https://lkml.org/lkml/2018/10/23/624

**Comment by xuliker** [comment]
* The linux devs can rescind their license grant.

```
The linux devs can rescind their license grant. Why don't they if they 
don't like the CoC. They did NOT give their code away. They merely 
licensed it to people for nothing. Licenses can be rescinded. The 
License text itself doesn't even disclaim the possibility of rescission. 
All it says is that those who are down stream of a violator won't have 
their license automatically revoked if the violator's is.

(No this was NOT "debunked", it's a complete lie to say that the owner 
of the copyright cannot rescind his freely licensed (no consideration 
paid) property: yes the free software people ARE LYING - but EVERYONE 
now believes their lie even though their debunking was immediately 
debunked itself. - Since none of you woman worshiping 
anti-marry-little-girls** faggots are lawyers you believe whatever 
you're told by "SFConservancy"*)

*(Who even had to bring in outside council themselves, inorder to 
conflate clause 4, having so little expertise on the subject in-house)

**(YHWH explicitly allows men to have female children as brides (Devarim 
chapter 22, verse 28 (na'ar) (same expressed in the Septuagint (padia)) 
(including in cases of a forceful taking (taphas)) (Yes, white men 
(americans, anglos) are the enemy of YHWH's law, worldwide championing a 
system that condemns all men to slavery at women's feet - the position 
they themselves naturally gravitate to)
```

* [PATCH] MAINTAINERS: Add x86 early-quirks.c file pattern to PCI

```
Bjorn Helgaas <bhelgaas@google.com>
arch/x86/kernel/early-quirks.c
```

* [PATCH] V6 init/main.c Enable watchdog_thresh control from kernel 

```c
Both graphics and serial consoles are exposed to hard lockups
when handling a large amount of messaging. The kernel watchdog_thresh
parameter up to now has not been available to be set on the kernel line for
early boot.
This patch allows the setting of watchdog_thresh to be increased
when needed to avoid the hard lockups in the console code.

Signed-off-by: Laurence Oberman <loberman@redhat.com>
Acked-by: Randy Dunlap <rdunlap@infradead.org>
---
 Documentation/admin-guide/kernel-parameters.txt |  7 +++++++
 init/main.c                                     | 11 +++++++++++
 2 files changed, 18 insertions(+)

diff --git a/Documentation/admin-guide/kernel-parameters.txt b/Documentation/admin-guide/kernel-parameters.txt
index 4cdcd1a..102382f 100644
--- a/Documentation/admin-guide/kernel-parameters.txt
+++ b/Documentation/admin-guide/kernel-parameters.txt
@@ -4932,6 +4932,13 @@
 			or other driver-specific files in the
 			Documentation/watchdog/ directory.
 
+	watchdog_thresh=
+			This parameter allows early boot to change the
+			value of the watchdog timeout threshold from the default
+			of 10 seconds to avoid hard lockups.  Example:
+			watchdog_thresh=30
+			Default: 10
+
 	workqueue.watchdog_thresh=
 			If CONFIG_WQ_WATCHDOG is configured, workqueue can
 			warn stall conditions and dump internal state to
diff --git a/init/main.c b/init/main.c
index 1c3f902..d5511a9 100644
--- a/init/main.c
+++ b/init/main.c
@@ -1038,6 +1038,17 @@ static int __init set_debug_rodata(char *str)
 __setup("rodata=", set_debug_rodata);
 #endif
 
+#ifdef CONFIG_LOCKUP_DETECTOR
+extern int watchdog_thresh;
+
+static int __init watchdog_thresh_setup(char *str)
+{
+	get_option(&str, &watchdog_thresh);
+	return 1;
+}
+__setup("watchdog_thresh=", watchdog_thresh_setup);
+#endif
+
 #ifdef CONFIG_STRICT_KERNEL_RWX
 static void mark_readonly(void)
 {
-- 
1.8.3.1

```

* [PATCH v14 0/3]: perf: reduce data loss when profiling highly
  * https://lkml.org/lkml/2018/10/15/169
* **[PATCH 00/10] steal tasks to improve CPU utilization**
  * https://lkml.org/lkml/2018/10/22/526
* [RFC PATCH v3] ftrace: support very early function tracing

```c
Hi Steve,
I believe this patch is quite stable. I know some folks that were using
it to debug early VM/KVM handshak and seemed very helpful for them.


Previous changes:

PATCH v1: Initial patch

PATCH v2:

   Removed arch specific code and use the default clock.
   Add more code re-usability
   Add HAVE_VERY_EARLY_FTRACE config option which is disabled by default


PATCH v3 starts here :
---------------------------------------------------------------------

The very early tracing will start from the beginning of start_kernel()
and will stop at ftrace_init()

start_kernel()
{
   ftrace_early_init() <--- start early function tracing
   ...
   (calls)
   ...
   ftrace_init()       <--- stop early function tracing
   early_trace_init(); <--- fill ring buffer, before starting function tracer
   ...
}

The events are placed in a temporary buffer, which will be copied to
the trace buffer after memory setup.

Dynamic tracing is not implemented with live patching, we use
ftrace_filter and ftrace_notrace to find which functions to be
filtered (traced / not traced), then during the callback from
mcount hook, we do binary search lookup to decide which function
to be save or not.

Signed-off-by: Abderrahmane Benbachir <abderrahmane.benbachir@polymtl.ca>
Cc: Steven Rostedt <rostedt@goodmis.org>
Cc: Ingo Molnar <mingo@redhat.com>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Cc: linux-kernel@vger.kernel.org
---
  arch/x86/Kconfig            |   1 +
  arch/x86/kernel/ftrace_32.S |  45 ++++--
  arch/x86/kernel/ftrace_64.S |  14 ++
  include/linux/ftrace.h      |  16 +-
  init/main.c                 |   1 +
  kernel/trace/Kconfig        |  51 +++++++
  kernel/trace/ftrace.c       | 289 +++++++++++++++++++++++++++++++++++-
  kernel/trace/trace.c        |   3 +
  8 files changed, 408 insertions(+), 12 deletions(-)
```

* [PATCH v1 2/2] x86/hyperv: make HvNotifyLongSpinWait hypercall
  * https://lkml.org/lkml/2018/10/19/624
* [PATCH RFC 1/1] drivers: pinctrl: qcom: add wakeup capability to
  * https://lkml.org/lkml/2018/10/11/169

```c
Hi all,

This is a second attempt at enabling wakeup using GPIO IRQs. On the QCOM
SDM845i SoC, even when the TLMM gpiochip is powered off the the PDC always-on
interrupt controller can sense the GPIO interrupt as a signal, wakeup the
interrupt and replay the interrupt at the GIC.

Since, the wakeup interrupt is replayed at the GIC and not at the TLMM, it
would have to be considered as a separate interrupt line and not in hierarchy
with the GPIO interrupt line which is chained to the GIC, using a single
summary line.

The earlier approach [1] was based on configuring a PDC interrupt whenever a
GPIO was requested as an IRQ and the PDC irq line was kept disabled. During
suspend and resume, the TLMM line was disabled and the PDC line was enabled.
While this works for suspend/resume, it becomes an additional complexity to
handle when entering a system low power mode where the TLMM could be powered
off when the CPUs are idle.

The approach here is to use PDC interrupt at all times (while keeping the GPIO
irq disabled). The PDC interrupt handler invokes the action handler of the GPIO
IRQ instead. This allows us to avoid interrupt hand-offs while entering idle
states and keeps things very simple. The wake_irq_gpio_handler() is the crux of
this new revision.

This submission does not include the DT bindings and associated documentation.
Once the core design is resolved I will send a complete set with dependencies
for this SoC. Please let me know if this approach would work or do you see any
issues that I may have missed.

Thanks,
Lina

[1]. https://lkml.org/lkml/2018/9/4/846

Lina Iyer (1):
  drivers: pinctrl: qcom: add wakeup capability to GPIO

 drivers/pinctrl/qcom/pinctrl-msm.c | 91 +++++++++++++++++++++++++++++-
 1 file changed, 90 insertions(+), 1 deletion(-)

```

* [PATCH V9 01/21] csky: Build infrastructure
  * https://lkml.org/lkml/2018/10/16/223
* [GIT PULL] libnvdimm for 4.20
  * https://lkml.org/lkml/2018/10/25/96

```c
Hi Linus, please pull from:

  git://git.kernel.org/pub/scm/linux/kernel/git/nvdimm/nvdimm tags/libnvdimm-for-4.20

...to receive the libnvdimm updates for this cycle. No interactions
with other trees this time around. There are updates to the core to
improve driver initialization time, and updates to the 'nfit' driver
for fixing ARS corner cases and exposing some more hardware attributes
to userspace. It has all been in -next for a few releases now with no
reported issues.
```

* [PATCH v37 0/3] Virtio-balloon: support free page reporting
  * https://lkml.org/lkml/2018/8/27/29

```c
The new feature, VIRTIO_BALLOON_F_FREE_PAGE_HINT, implemented by this
series enables the virtio-balloon driver to report hints of guest free
pages to host. It can be used to accelerate virtual machine (VM) live
migration. Here is an introduction of this usage:

Live migration needs to transfer the VM's memory from the source machine
to the destination round by round. For the 1st round, all the VM's memory
is transferred. From the 2nd round, only the pieces of memory that were
written by the guest (after the 1st round) are transferred. One method
that is popularly used by the hypervisor to track which part of memory is
written is to have the hypervisor write-protect all the guest memory.

This feature enables the optimization by skipping the transfer of guest
free pages during VM live migration. It is not concerned that the memory
pages are used after they are given to the hypervisor as a hint of the
free pages, because they will be tracked by the hypervisor and transferred
in the subsequent round if they are used and written.
```

* [PATCH] kvm/x86 : remove unused struct definition

```c
structure svm_init_data is never used. So remove it.

Signed-off-by: Peng Hao <peng.hao2@zte.com.cn>
---
 arch/x86/kvm/svm.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/arch/x86/kvm/svm.c b/arch/x86/kvm/svm.c
index 61ccfb1..5c7dc8b 100644
--- a/arch/x86/kvm/svm.c
+++ b/arch/x86/kvm/svm.c
@@ -675,11 +675,6 @@ struct svm_cpu_data {
 
 static DEFINE_PER_CPU(struct svm_cpu_data *, svm_data);
 
-struct svm_init_data {
-	int cpu;
-	int r;
-};
-
 static const u32 msrpm_ranges[] = {0, 0xc0000000, 0xc0010000};
 
 #define NUM_MSR_MAPS ARRAY_SIZE(msrpm_ranges)
```

* [PATCH] tools/kvm_stat: switch to python3
  * https://lkml.org/lkml/2017/10/3/1059
* SSL Expired

```
Hello,

Please forward to the relevant department/organisation/person.

The SSL certificate for https://www.linux-kvm.org/ has expired and
needs renewing.

Kind regards
Daniel Journo
```

* [PATCH 4/4] kvm, vmx: remove manually coded vmx instructions
  * https://lkml.org/lkml/2018/10/24/356
* [PATCH v8 0/3] x86/boot/KASLR: Parse ACPI table and limit kaslr
  * https://lkml.org/lkml/2018/10/10/343
* [PATCH] x86/ioremap: tighten integer overflow checking
  * https://lkml.org/lkml/2018/10/25/232

```c
The current check is a bit off in the case where "phys_addr + size"
wraps to zero because then "last_addr" is set to ULONG_MAX which is >=
phys_addr.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
---
 arch/x86/mm/ioremap.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/mm/ioremap.c b/arch/x86/mm/ioremap.c
index 5378d10f1d31..ee43df3ebe66 100644
--- a/arch/x86/mm/ioremap.c
+++ b/arch/x86/mm/ioremap.c
@@ -146,9 +146,9 @@ static void __iomem *__ioremap_caller(resource_size_t phys_addr,
 	void __iomem *ret_addr;
 
 	/* Don't allow wraparound or zero size */
-	last_addr = phys_addr + size - 1;
-	if (!size || last_addr < phys_addr)
+	if (!size || phys_addr + size < phys_addr)
 		return NULL;
+	last_addr = phys_addr + size - 1;
 
 	if (!phys_addr_valid(phys_addr)) {
 		printk(KERN_WARNING "ioremap: invalid physical address %llx\n",
-- 
2.11.0
```

* [PATCH v3] kernel/signal: Signal-based pre-coredump notification
  * https://lkml.org/lkml/2018/10/13/47
* [PATCH] MAINTAINERS: add myself as co-maintainer of gpiolib
  * https://lkml.org/lkml/2018/10/24/634

```c
From: Bartosz Golaszewski <bgolaszewski@baylibre.com>

As discussed with Linus Walleij - I'm adding myself as
the co-maintainer.

Cc: Benoît Cousson <bcousson@baylibre.com>
Cc: Kevin Hilman <khilman@baylibre.com>
Cc: Michael Turquette <mturquette@baylibre.com>
Signed-off-by: Bartosz Golaszewski <bgolaszewski@baylibre.com>
---
 MAINTAINERS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index af2a61a31a8e..facf6a859854 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -6279,6 +6279,7 @@ F:	tools/testing/selftests/gpio/
 
 GPIO SUBSYSTEM
 M:	Linus Walleij <linus.walleij@linaro.org>
+M:	Bartosz Golaszewski <bgolaszewski@baylibre.com>
 L:	linux-gpio@vger.kernel.org
 T:	git git://git.kernel.org/pub/scm/linux/kernel/git/linusw/linux-gpio.git
 S:	Maintained
```

**Comment by xuliker** [comment]
* [GIT PULL] Driver core patches for 4.20-rc1
* Re: [PATCH 00/37] perf/core improvements and fixes
* [PATCH v4 2/2] perf/x86/kvm: Avoid unnecessary work in guest filtering
* [PATCH v4 1/2] x86/cpufeature: Add facility to match microcode revisions
* Re: [PATCH v2] x86/microcode: Handle negative microcode revisions
* Re: More checks for patterns? was: Fix pattern handling optimalization
* Re: [PATCH] mm/page_owner: use vmalloc instead of kmalloc
* [PATCH v3] x86/numa_emulation: Fix uniform-split numa emulation
* [GIT PULL] Please pull RDMA subsystem changes
* **[PATCH 3/3] kernel/workqueue: Suppress a false positive lockdep**
* Re: [v3 03/12] x86/fsgsbase/64: Add intrinsics/macros for FSGSBASE instructions
* Re: Linux 4.19
* [RESEND PATCH v2] ttyprintk: make the printk log level configurable
* Allow persistent memory to be used like normal RAM

**Comment by xuliker** [comment]
* [RFC 00/60] Coscheduling for Linux
* A concern about overflow ring buffer mode
* [git pull] IOMMU Updates for Linux v4.20
* [PATCH 2/4] migration: introduce lockless multithreads model
* [PATCH 2/4] kvm, vmx: move register clearing out of assembly path
* [PATCH v4 0/5] ia64: system call table generation support
* [GIT PULL] C-SKY(csky) Port for Linux 4.20 
* [PATCH v5 00/13] arch/x86: AMD QoS support

**Comment by xuliker** [comment]
```
* [PATCH V5 0/5] KVM: X86: Introducing ROE Protection Kernel
* [PATCH 2/4] migration: introduce lockless multithreads model
* [for-next][PATCH 00/18] tracing: For this merge window
*  Which SPDX Identifier for files without explicit GPL version
```

**Comment by xuliker** [comment]
- [ANNOUNCE] v4.18.16-rt9
- Re: x86/paravirt: Use a single ops structure
- [LINUX PATCH v11 0/3] Add support for Arasan NAND Flash

**Comment by xuliker** [comment]
* [RFC PATCH] Implement /proc/pid/kill
* [PATCH] fs/proc: introduce /proc/stat2 file
* [RFC] Proposal for FMC Subsystem Eradication
* [RFC PATCH v2] Minimal non-child process exit notification support
* [RFC 00/60] Coscheduling for Linux
* [PATCH 08/10] tools power turbostat: Override CFLAGS assignments
* [PATCH] Choose CPU based on allocated IRQs
* Can VFIO pin only a specific region of guest mem when use pass
* [PATCH v13 08/12] KVM: x86: Add Intel PT context switch for each vcpu
* x86/paravirt: Use a single ops structure
* [RFC 00/60] Coscheduling for Linux
* [PATCH kvm-unit-tests] scripts: Speedup MAX_SMP check
* [RFC PATCH] kvm: Use huge pages for DAX-backed files
* [RFC 00/20] ns: Introduce Time Namespace
* [PATCH v2] bit_spinlock: introduce smp_cond_load_relaxed
* [PATCH v4 1/2] virtio-gpu: add VIRTIO_GPU_F_EDID feature
* [PATCH net] vhost: Fix Spectre V1 vulnerability
* [PATCH] kbuild: remove unused cc-fullversion variable
*  [PATCH] kbuild: consolidate single targets
* [Bug 201565] Intel Processor Trace enabling in KVM
* Trying to build KVM
* Opcodes instead of mnemonics for VMX instructions
* [GIT PULL] More ACPI updates for v4.20-rc1
* [PATCH v5 00/15] Add utilization clamping support
* [PATCH v5 0/8] Introduce STPMIC1 PMIC Driver
* [PATCH v4 2/2] drm/virtio: add edid support

**Comment by xuliker** [comment]
* [PATCH][PATCH V7] watchdog/core: Add watchdog_thresh command line  parameter
* Re: [PATCH 01/21] of: Add cpu node iterator for_each_of_cpu_node()
* [PATCH RESEND] initramfs: cleanup incomplete rootfs
* Re: [RFC][PATCH v1] ACPI / scan: Create platform device for BOSC0200
* Re: [PATCH 11/14] irq: add support for allocating (and affinitizing)
* [PULL 0/7] x86 queue, 2018-10-30
* Re: [PATCH 2/5] drm/virtio: add uapi for in and out explicit fences
* Re: [PATCH 4/4] kvm, vmx: remove manually coded vmx instructions
* [PATCH] arm64/numa: Add more vetting in numa_set_distance()

**Comment by xuliker** [comment]
* [PATCH v3 00/20] vmw_balloon: compaction, shrinker, 64-bit, etc.
* RFC PATCH] Implement /proc/pid/kill
* [PATCH v2] proc: use ns_capable instead of capable for timerslack_ns
* [PATCH] Choose CPU based on allocated IRQs
* [GIT PULL] VFIO updates for v4.20-rc1
* [PATCH V5 0/5] KVM: X86: Introducing ROE Protection Kernel
* [RFC PATCH] kvm: Use huge pages for DAX-backed files
* [PULL 0/7] x86 queue, 2018-10-30
* [PATCH] kvm: x86: Don't modify MSR_PLATFORM_INFO on vCPU reset
* Should KVM creation fail if debugfs fails?
* [RFC] rcu: doc: update example about stale data
* [PATCH 0/5] virgl: fence fd support
* [PATCH 0/4] Improve virtio ID allocation

**Comment by xuliker** [comment]
* steal tasks to improve CPU utilization
* watchdog/core: Add watchdog_thresh command line parameter
* [RFC PATCH] kvm: Use huge pages for DAX-backed files
* [GIT PULL] percpu changes for v4.20-rc1
* [PATCH] x86/build: Build VSMP support only if CONFIG_PCI is
* [kvm PATCH v5 2/4] kvm: x86: Dynamically allocate guest_fpu
* [PATCH v1 0/8] Intel Virtual PMU Optimization
* [PATCH V5 0/5] KVM: X86: Introducing ROE Protection Kernel
* [PATCH v5] printk: Add line-buffered printk() API.
* [PATCH v5 00/13] arch/x86: AMD QoS support

**Comment by xuliker** [comment]
* [PATCH v3] genirq/matrix: Choose CPU for managed IRQs based on how many of them are allocated
* [PATCH v2 00/11] arch/x86: AMD QoS support
* PLEASE READ: kernelci.org bot email problems
* [Qemu-devel] [QEMU PATCH v2 0/2]: KVM: i386: Add support for save
* RFC: userspace exception fixups
* [PULL] vhost: cleanups and fixes
* [PULL 0/7] x86 queue, 2018-10-30
* [RFC PATCH v8 0/4] Media Device Allocator API
* No interrupt was generated using MSI

**Comment by xuliker** [comment]
* [PATCH 4.19 09/24] vhost: Fix Spectre V1 vulnerability
* [PATCH 4.18 087/150] KVM: x86: support CONFIG_KVM_AMD=y with CONFIG_CRYPTO_DEV_CCP_DD=m
* [PATCH v8 0/4] KASLR feature to randomize each loadable module
* [PATCH v15 00/23] Intel SGX1
* [PATCH v15 02/23] x86/cpufeatures: Add Intel-defined SGX feature
* [PATCH 0/1] vhost: add vhost_blk driver
* [PATCH v3 1/2] kretprobe: produce sane stack

**Comment by xuliker** [comment]
* [GIT PULL] x86 fixes
* [PATCH 0/2] perf intel-pt: Add more information to debug log
* Linux Foundation Technical Advisory Board Elections -- Call for
* [PATCH v4 1/8] iommu: Add APIs for multiple domains per device
* [PATCH RFC] hist lookups
* Linux 4.20-rc1
* [PATCH] s390: numa: Export __node_distance
* [PATCH v2 00/12] tracing: Unifying dynamic event interface
* [PATCH v1 1/8] perf/x86: add support to mask counters from host
* [PATCH v4 00/12] iommu/vt-d: Add scalable mode support
* [PATCH V3 0/3] Unmap EFI boot services code/data regions after boot.

**Comment by xuliker** [comment]
* [PATCH] x86/cpu: Avoid endless loop to get the number of cache leaves
* [PATCH v11 00/26] Speculative page faults
* [PATCH 0/1] vhost: add vhost_blk driver
* [PATCH v15 05/23] x86/cpu/intel: Detect SGX support and update
* [PATCH v2] Document /proc/pid PID reuse behavior
* [PATCH] init/Kconfig: make CC_OPTIMIZE_FOR_DEBUGGING visible only for GCC
* [PATCH v6 0/5] LSM: Support ptrace sidechannel access checks
* [PATCH] docs: Extend trusted keys documentation for TPM 2.0
* [PATCH RFC] hist lookups
* Build regressions/improvements in v4.20-rc1
* [PATCH 0/1] vhost: add vhost_blk driver
* [RFC PATCH v4 01/13] ktask: add documentation
* [RFC PATCH v4 00/13] ktask: multithread CPU-intensive kernel work
* [driver-core PATCH v5 0/9] Add NUMA aware async_schedule calls
* [mm PATCH v5 0/7] Deferred page init improvements
